annet/mesh/basemodel.py

```diff
@@ -178,6 +181,43 @@ def merge(first: Any, /, *others: Any) -> Any:
     return first
 
 
+def merge_dataclass(a: Any, b: Any) -> Any:
+    if a.__class__ != b.__class__:
+        raise MergeForbiddenError(
+            f"Dataclasses belonging to different instaces can't be merged:\n"
+            f"Old: {a}\n"
+            f"New: {b}"
+        )
+
+    empty = a.__class__()
```

Comment:
it works only if all fields have default. Should we consider processing each field default separately?